android/expoview/src/main/java/versioned/host/exp/exponent/modules/internal/ExponentAsyncStorageModule.java

```diff
@@ -15,22 +15,22 @@
 
 import expo.modules.updates.manifest.raw.RawManifest;
 import host.exp.exponent.ExponentManifest;
+import host.exp.exponent.kernel.ExperienceKey;
 import host.exp.exponent.kernel.KernelProvider;
 
 @ReactModule(name = ExponentAsyncStorageModule.NAME, canOverrideExistingModule = true)
 public class ExponentAsyncStorageModule extends AsyncStorageModule {
 
-  public static String experienceIdToDatabaseName(String experienceId) throws UnsupportedEncodingException {
-    String experienceIdEncoded = URLEncoder.encode(experienceId, "UTF-8");
-    return "RKStorage-scoped-experience-" + experienceIdEncoded;
+  public static String experienceKeyToDatabaseName(ExperienceKey experienceKey) throws UnsupportedEncodingException {
+    return "RKStorage-scoped-experience-" + experienceKey.getUrlEncodedScopeKey();
```

Comment:
Are we confident this isn't used in standalone (turtle v1) apps? If it is, we need to use the stable legacy ID here.